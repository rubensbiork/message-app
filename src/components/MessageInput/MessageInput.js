import React, { PureComponent } from 'react';
import ReactDOM from 'react-dom';
import PropTypes from 'prop-types';

const ENTER_KEY_CODE = 13;
const ESC_KEY_CODE = 27;

export default class MessageInput extends PureComponent {
  state = {
    isEditing: false,
    text: this.props.existingValue || '',
  };

  componentDidMount() {
    ReactDOM.findDOMNode(this).focus();
  }
  _commitChanges = () => {
    const {
      text,
    } = this.state;

    if (text.trim() !== '') {
      this.props.onSave(text);
      this.setState({text: ''});
    }
  };

  _onChange = (e) => {
    this.setState({text: e.target.value});
  };
  _onKeyDown = (e) => {
    if (e.keyCode === ESC_KEY_CODE) {
      if (this.props.onCancel) {
        this.props.onCancel();
      }
    }
    if (e.keyCode === ENTER_KEY_CODE) {
      this._commitChanges();
    }
  };
  render() {
    return (
      <input
        className={this.props.className}
        onChange={this._onChange}
        onKeyDown={this._onKeyDown}
        placeholder={this.props.placeholder}
        value={this.state.text}
      />
    );
  }
}

MessageInput.defaultProps = {
  className: '',
  existingValue: '',
  onCancel: () => {},
  onSave: () => {},
  placeholder: '',
};

MessageInput.propTypes = {
  className: PropTypes.string,
  existingValue: PropTypes.string,
  onCancel: PropTypes.func,
  onSave: PropTypes.func.isRequired,
  placeholder: PropTypes.string,
};
