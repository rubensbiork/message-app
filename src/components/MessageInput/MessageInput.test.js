import React from 'react';
import { shallow } from 'enzyme';

import MessageInput from './MessageInput';

const wrapper = props => shallow(<MessageInput {...props} />);

describe('components/<MessageInput />', () => {
  it('Should render', () => {
    const component = wrapper({});

    expect(
      component,
    ).toMatchSnapshot();
  });
});
