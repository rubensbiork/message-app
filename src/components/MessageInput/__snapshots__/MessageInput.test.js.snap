// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/<MessageInput /> Should render 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <input
    className=""
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder=""
    value=""
/>,
  "nodes": Array [
    <input
      className=""
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder=""
      value=""
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 1,
      "_context": Object {},
      "_currentElement": <MessageInput
        className=""
        existingValue=""
        onCancel={[Function]}
        onDelete={[Function]}
        onSave={[Function]}
        placeholder=""
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": MessageInput {
        "_commitChanges": [Function],
        "_onChange": [Function],
        "_onKeyDown": [Function],
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "className": "",
          "existingValue": "",
          "onCancel": [Function],
          "onDelete": [Function],
          "onSave": [Function],
          "placeholder": "",
        },
        "refs": Object {},
        "state": Object {
          "isEditing": false,
          "text": "",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <input
          className=""
          onChange={[Function]}
          onKeyDown={[Function]}
          placeholder=""
          value=""
/>,
        "_debugID": 2,
        "_renderedOutput": <input
          className=""
          onChange={[Function]}
          onKeyDown={[Function]}
          placeholder=""
          value=""
/>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <MessageInput
    className=""
    existingValue=""
    onCancel={[Function]}
    onDelete={[Function]}
    onSave={[Function]}
    placeholder=""
/>,
}
`;
